x/crosschain/keeper/msg_server_vote_inbound_tx.go

```diff
@@ -188,22 +188,6 @@ func (k msgServer) VoteOnObservedInboundTx(goCtx context.Context, msg *types.Msg
 				return k.UpdateNonce(tmpCtx, chain.ChainId, &cctx)
 			}()
 			if err != nil {
-				// do not commit anything here as the CCTX should be aborted
-
-				// gas payment for erc20 type might fail because no liquidity pool is defined to swap the zrc20 token into the gas token
-				// in this gas we should refund the sender on ZetaChain
-				if cctx.InboundTxParams.CoinType == common.CoinType_ERC20 {
-
-					if err := k.RefundAmountOnZetaChain(ctx, cctx, cctx.InboundTxParams.Amount); err != nil {
-						// log the error
-						k.Logger(ctx).Error("failed to refund amount of aborted cctx on ZetaChain",
-							"error", err,
-							"sender", cctx.InboundTxParams.Sender,
-							"amount", cctx.InboundTxParams.Amount.String(),
-						)
-					}
```

Comment:
I think we should keep the auto-refund for ERC20.
The reason is that unlike other cases like Zeta that implies interacting with an app that doesn't correctly implement the interface, the aborted case can happen for protocol reasons for ERC20, if a refund must be executed and the liquidity in the pools is not enough to swap for gas. This is especially true if the liquidity cap for the ZRC20 is reached and the CCTX are automatically reverted.

Why not calling `RefundAmountOnZetaChainERC20` in that case to provide directly the address if we want to keep the `RefundAmountOnZetaChain` logic simpler?

Reply:
I don't think it matters which function we use, the refund address gets passed down directly. 

however, having the admin TX enables us to monitor the pools before we send in the TX, and we can choose not to refund if we don't want to.